src/select/select.js

```diff
@@ -256,7 +261,12 @@ class Select extends React.Component<PropsT, StatelessStateT> {
           InputContainer: InputContainer,
           After: () => (
             <SearchIcon
-              onClick={e => this.onChange(e, STATE_CHANGE_TYPE.clearAll)}
+              disabled={disabled}
+              onClick={e => {
+                if (!disabled) {
```

Comment:
if all Input component were disabled, we wouldn't need to add this condition everywhere as event handlers shouldn't be called